include/verilated_types.h

```diff
@@ -1003,12 +1004,158 @@ std::string VL_TO_STRING(const VlUnpacked<T_Value, T_Depth>& obj) {
     return obj.to_string();
 }
 
+class VlClass;  // See below
+
+//===================================================================
+// Class providing delayed deletion of garbage objects. Objects get deleted only when 'deleteAll()'
+// is called, or the deleter itself is destroyed.
+
+class VlDeleter final {
+    // MEMBERS
+    std::vector<VlClass*> m_newGarbage;  // Queue of new objects to delete
+    std::vector<VlClass*> m_toDelete;  // Queue of objects currently being deleted
+    VerilatedMutex m_mutex;  // Mutex protecting the 'new garbage' queue
```

Comment:
```suggestion
    std::vector<VlClass*> m_newGarbage VL_GUARDED_BY(m_mutex);  // Queue of new objects to delete
    mutable VerilatedMutex m_mutex;  // Mutex protecting the 'new garbage' queue
```

toDelete can now be function local.

Reply:
Point accepted.  Then I think m_toDelete also needs VL_GUARDED_BY(m_mutex).